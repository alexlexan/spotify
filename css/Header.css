.header {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  margin-bottom: 30px;
  height: 44px;
}

.header__right {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.header__left {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-flex: 0.5;
      -ms-flex: 0.5 1;
          flex: 0.5 1;
}

.header__arrowLeft {
  font-size: 2.1rem;
}

.header__arrowLeft:hover {
  cursor: pointer;
}

.header__arrowRight {
  font-size: 2.1rem;
  cursor: pointer;
}

.header__search {
  margin-left: 20px;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  background-color: white;
  padding: 10px;
  border-radius: 30px;
  color: gray;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.header__search > input {
  border: none;
  width: 100%;
}

.header__right > h4 {
  margin-left: 10px;
}
/*# sourceMappingURL=Header.css.map */