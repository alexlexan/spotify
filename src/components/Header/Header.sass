.header 
  display: flex
  justify-content: space-between
  margin-bottom: 30px
  height: 44px


.header__right 
  display: flex
  align-items: center

.header__left
  display: flex
  align-items: center  
  flex: 0.5 1

.header__arrowLeft
  font-size: 2.1rem
  &:hover
    cursor: pointer 

.header__arrowRight
  font-size: 2.1rem  
  &:hover
  cursor: pointer  

.header__search 
  margin-left: 20px
  width: 100%
  display: flex
  background-color: white
  padding: 10px
  border-radius: 30px
  color: gray
  align-items: center


.header__search > input 
  border: none
  width: 100%


.header__right > h4 
  margin-left: 10px
