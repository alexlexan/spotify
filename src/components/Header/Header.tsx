import React from "react";
import "../Header/Header.sass";
import { Avatar } from "@material-ui/core";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import { AppState } from "../../store/rootReducer";

const Header = () => {
  const user = useSelector((state: AppState) => state.spotify.user);
  const history = useHistory();

  const Back = () => {
    history.location.key && history.goBack();
  };
  const Forward = () => {
    history.goForward();
  };

  return (
    <div className="header">
      <div className="header__left">
        <ArrowBackIosIcon
          className="header__arrowLeft"
          onClick={() => Back()}
        />
        <ArrowForwardIosIcon
          className="header__arrowRight"
          onClick={() => Forward()}
        />
      </div>
      {user && (
        <div className="header__right">
          <Avatar
            alt={user.display_name}
            src={(user.images as SpotifyApi.ImageObject[])[0]?.url}
          />
          <h4>{user.display_name}</h4>
        </div>
      )}
    </div>
  );
};

export default React.memo(Header);
