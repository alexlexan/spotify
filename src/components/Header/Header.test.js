import React from "react";
import Header from "./Header";
import { useSelector } from "react-redux";
import { shallow } from "enzyme";
import { useHistory } from "react-router-dom";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";

jest.mock("react-router-dom", () => ({
  useHistory: jest.fn(),
}));

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
}));

describe("test Header component", () => {
  let component, user, history;
  const mockGoForward = jest.fn();
  const mockGoBack = jest.fn();
  user = {
    display_name: "user",
    images: [{ url: "imagesUrl" }],
  };

  history = {
    location: {
      key: "history",
    },
    goBack: mockGoBack,
    goForward: mockGoForward,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("render Header with user", () => {
    useSelector.mockImplementation((selector) =>
      selector({ spotify: { user } })
    );
    component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });

  it("render Header without user", () => {
    useSelector.mockImplementation((selector) =>
      selector({ spotify: { user: null } })
    );
    component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });

  it("should be call mockGoBack and mockGoForward", () => {
    useSelector.mockImplementation((selector) =>
      selector({ spotify: { user } })
    );
    useHistory.mockReturnValue(history);
    component = shallow(<Header />);
    expect(component).toMatchSnapshot();

    component.find(ArrowBackIosIcon).simulate("click");
    expect(mockGoBack).toBeCalledTimes(1);

    component.find(ArrowForwardIosIcon).simulate("click");
    expect(mockGoForward).toBeCalledTimes(1);
  });

  it("should dont call mockGoBack", () => {
    useSelector.mockImplementation((selector) =>
      selector({ spotify: { user } })
    );
    useHistory.mockReturnValue({
      location: { key: null },
      goBack: mockGoBack,
      goForward: mockGoForward,
    });
    component = shallow(<Header />);
    expect(component).toMatchSnapshot();

    component.find(ArrowBackIosIcon).simulate("click");
    expect(mockGoBack).toBeCalledTimes(0);
  });
});
