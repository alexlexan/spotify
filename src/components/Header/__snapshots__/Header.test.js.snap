// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header component render Header with user 1`] = `
<div
  className="header"
>
  <div
    className="header__left"
  >
    <ArrowBackIosIcon
      className="header__arrowLeft"
      onClick={[Function]}
    />
    <ArrowForwardIosIcon
      className="header__arrowRight"
      onClick={[Function]}
    />
  </div>
  <div
    className="header__right"
  >
    <WithStyles(ForwardRef(Avatar))
      alt="user"
      src="imagesUrl"
    />
    <h4>
      user
    </h4>
  </div>
</div>
`;

exports[`test Header component render Header without user 1`] = `
<div
  className="header"
>
  <div
    className="header__left"
  >
    <ArrowBackIosIcon
      className="header__arrowLeft"
      onClick={[Function]}
    />
    <ArrowForwardIosIcon
      className="header__arrowRight"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`test Header component should be call mockGoBack and mockGoForward 1`] = `
<div
  className="header"
>
  <div
    className="header__left"
  >
    <ArrowBackIosIcon
      className="header__arrowLeft"
      onClick={[Function]}
    />
    <ArrowForwardIosIcon
      className="header__arrowRight"
      onClick={[Function]}
    />
  </div>
  <div
    className="header__right"
  >
    <WithStyles(ForwardRef(Avatar))
      alt="user"
      src="imagesUrl"
    />
    <h4>
      user
    </h4>
  </div>
</div>
`;

exports[`test Header component should dont call mockGoBack 1`] = `
<div
  className="header"
>
  <div
    className="header__left"
  >
    <ArrowBackIosIcon
      className="header__arrowLeft"
      onClick={[Function]}
    />
    <ArrowForwardIosIcon
      className="header__arrowRight"
      onClick={[Function]}
    />
  </div>
  <div
    className="header__right"
  >
    <WithStyles(ForwardRef(Avatar))
      alt="user"
      src="imagesUrl"
    />
    <h4>
      user
    </h4>
  </div>
</div>
`;
